src/components/pagination/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useMemo, useState } from 'react'
+import {
+  StandaloneChevronLeftRegularIcon,
+  StandaloneChevronRightRegularIcon,
+} from '@deriv/quill-icons/Standalone'
+import qtMerge from 'qtMerge'
+import { usePaginationRange } from 'hooks/usePaginationRange'
+import PaginationButton from './pagination-button'
+import { paginationVariants } from './pagination.classnames'
+import { PaginationProps } from './pagination.types'
+
+/**
+ * Component that can divide large lists of data into smaller more manageable pages.
+ * @name Pagination
+ * @param className - Styles to be applied to Pagination body
+ * @param contentPerPage - Number of items to be displayed per page
+ * @param contentLength - Total length of data to be paginated
+ * @param onPagination - Function to handle pagination event. It passes an object with currentPage and totalPageCount
+ * @param variant - Variant of Pagination.
+ *
+ * @example
+ * <Pagination className="pagination__container" contentPerPage={10}
+ * contentLength={100} variant="number" onPagination={handlePagination}>
+ * />
+ */
+const Pagination = ({
+  className,
+  contentPerPage,
+  contentLength,
+  onPagination,
+  variant = 'number',
+}: PaginationProps) => {
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const totalPageCount = useMemo(() => {
+    const dataToDisplay = contentPerPage > 0 ? contentPerPage : 1
+    return Math.ceil(contentLength / dataToDisplay)
+  }, [contentLength, contentPerPage])
+
+  const paginationRange = usePaginationRange({
+    totalPageCount,
+    currentPage,
+    variant,
+  })
+
+  useEffect(() => {
+    onPagination({ currentPage, totalPageCount })
+  }, [currentPage, onPagination, totalPageCount])
+
+  const goToNextPage = () => setCurrentPage((page) => page + 1)
+
+  const gotToPreviousPage = () => setCurrentPage((page) => page - 1)
+
+  const changePage = (event: string | null) => {
+    const pageNumber = Number(event)
+    setCurrentPage(pageNumber)
+  }
+
+  return (
+    <section
+      className={qtMerge('flex items-center justify-center gap-400', className)}
```

Comment:
Make it `nav` tag and add `aria-label` and `role` to make it accessible. 🙏

Reply:
Please check this 🙏  [https://www.a11ymatters.com/pattern/pagination/](https://www.a11ymatters.com/pattern/pagination/)